nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -225,8 +225,12 @@ in
               esac
             done
             ''] ++ (map (pool: ''
-            echo "importing root ZFS pool \"${pool}\"..."
-            zpool import -d ${cfgZfs.devNodes} -N $ZFS_FORCE "${pool}"
+            echo -n "importing root ZFS pool \"${pool}\"..."
+            for t in 1 2 4 8; do
+              if zpool import -d ${cfgZfs.devNodes} -N $ZFS_FORCE "${pool}"; then break; fi
+              echo "Retrying in $t seconds"
```

Comment:
For debugging purposes, it might be good to have something like:

```bash
if [ $t -eq 8 ]; then
  echo "Failed to import ZFS dev nodes ${cfgZfs.devNodes} for the ${pool} pool in 4 tries."
else
  echo "Retrying in $t seconds"
  sleep $t
fi
```